pulsar-client/src/main/java/org/apache/pulsar/client/impl/BinaryProtoLookupService.java

```diff
@@ -92,15 +100,43 @@ public void updateServiceUrl(String serviceUrl) throws PulsarClientException {
      * @return broker-socket-address that serves given topic
      */
     public CompletableFuture<Pair<InetSocketAddress, InetSocketAddress>> getBroker(TopicName topicName) {
-        return findBroker(serviceNameResolver.resolveHost(), false, topicName, 0);
+        final MutableObject<CompletableFuture> newFutureCreated = new MutableObject<>();
```

Comment:
Maybe a simple `CompletableFuture` is enough, e.g.:
```
CompletableFuture<Pair<InetSocketAddress, InetSocketAddress>> localFuture = null;
```
All the `newFutureCreated.getValue()` could be replaced by `localFuture`.  It will helpful for jvm minor GC.

Reply:
No, we need to call the variable `localFuture` in the lambda expression, so it should be modified by `final`.

Reply:
OK~  I see.  Thank you:)